### PR TITLE
Add io.frama.tractor.carburetor

### PR DESCRIPTION
**Pull request by danialbehzadi**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

File: io.frama.tractor.carburetor.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+        --prefix=${FLATPAK_DEST} . --no-build-isolation
+      - mkdir -p ${FLATPAK_DEST}/share/appdata
+      - cp ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor.metainfo.xml ${FLATPAK_DEST}/share/appdata
+      - mkdir -p ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps
+      - cp ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps
+      - mkdir -p ${FLATPAK_DEST}/share/icons/hicolor/symbolic/apps
+      - cp ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor-symbolic.svg ${FLATPAK_DEST}/share/icons/hicolor/symbolic/apps
+      - mkdir -p ${FLATPAK_DEST}/share/applications
+      - cp ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor.desktop ${FLATPAK_DEST}/share/applications
```
Comment (hfiguiere) [inline io.frama.tractor.carburetor.yaml:60]:
```suggestion
      - install -Dm644 ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo
      - install -Dm644 ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor.svg -t ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps
      - install -Dm644 ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor-symbolic.svg -t ${FLATPAK_DEST}/share/icons/hicolor/symbolic/apps
      - install -Dm644 ${PWD}/src/carburetor/desktop/io.frama.tractor.carburetor.desktop -t ${FLATPAK_DEST}/share/applications
```

File: io.frama.tractor.carburetor.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      - type: archive
+        url: https://github.com/torproject/tor/archive/refs/tags/tor-0.4.7.13.tar.gz
+        sha256: a17b0e085a9927dffd414ed54b0c0f16caf1e8efdaba72878575c005c67856a1
+  - name: tractor
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+        --prefix=${FLATPAK_DEST} . --no-build-isolation
+      - mkdir -p ${FLATPAK_DEST}/share/glib-2.0/schemas
+      - cp ${PWD}/tractor/tractor.gschema.xml ${FLATPAK_DEST}/share/glib-2.0/schemas
```
Comment (hfiguiere) [inline io.frama.tractor.carburetor.yaml:41]:
```suggestion
      - install -Dm644 tractor/tractor.gschema.xml -t ${FLATPAK_DEST}/share/glib-2.0/schemas
```

**Comment by hfiguiere** [comment]
bot, build io.frama.tractor.carburetor

File: io.frama.tractor.carburetor.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        url: https://github.com/torproject/tor/archive/refs/tags/tor-0.4.7.13.tar.gz
+        sha256: a17b0e085a9927dffd414ed54b0c0f16caf1e8efdaba72878575c005c67856a1
+  - name: tractor
+    buildsystem: simple
+    build-commands:
+      - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}"
+        --prefix=${FLATPAK_DEST} . --no-build-isolation
+      - install -Dm644 tractor/tractor.gschema.xml -t ${FLATPAK_DEST}/share/glib-2.0/schemas
+      - glib-compile-schemas ${FLATPAK_DEST}/share/glib-2.0/schemas
+      - cp ${PWD}/tractor/SampleBridges ${FLATPAK_DEST}/lib/python3.11/site-packages/tractor/
```
Comment (bbhtt) [inline io.frama.tractor.carburetor.yaml:42]:
```suggestion
      - cp ${PWD}/tractor/SampleBridges ${FLATPAK_DEST}/lib/python$(python -c 'import sys; print("%s.%s" %sys.version_info[0:2])')/site-packages/tractor/
```

This won't break in between runtime bumps.

File: io.frama.tractor.carburetor.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
+  # DRI access for OpenGL rendering
+  - --device=dri
+  # Needs to talk to the network:
+  - --share=network
+  # Access to GNOME Virtual FileSystem
+  - --filesystem=xdg-run/gvfsd
+modules:
```
Comment (bbhtt) [inline io.frama.tractor.carburetor.yaml:18]: Add cleanups:

```
- "*.a"
- "*.la"
- /include
- /lib/pkgconfig
```

If torrc.sample isn't needed in `/etc`, add that too.

**Comment by bbhtt** [comment]
> Build [83149 successful](https://buildbot.flathub.org/#/builders/6/builds/83149) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/65784/io.frama.tractor.carburetor.flatpakref
> ```

It seems to get stuck at 5% or 10% in the bootstrapping stage.

**Comment by danialbehzadi** [comment]
> It seems to get stuck at 5% or 10% in the bootstrapping stage.

No such problem here. Please make sure there is no limitations for connecting to TOR on your network or use some working bridges.

**Comment by bbhtt** [comment]
> > It seems to get stuck at 5% or 10% in the bootstrapping stage.
> 
> No such problem here. Please make sure there is no limitations for connecting to TOR on your network or use some working bridges.

Tor on my system works fine.

bot, build

**Comment by bbhtt** [comment]
bot, build io.frama.tractor.carburetor

**Comment by bbhtt** [comment]
> Build [83186 successful](https://buildbot.flathub.org/#/builders/6/builds/83186) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/65820/io.frama.tractor.carburetor.flatpakref
> ```

Works now, I think I had some network hitch...

**Comment by danialbehzadi** [comment]
Any extra steps needed?

**Comment by bbhtt** [comment]
> Any extra steps needed?

No seems fine. Wait for a bit and a repo will be created.

**Pull request closed without merging**